Mathlib/RingTheory/Noetherian.lean

```diff
@@ -286,6 +286,23 @@ instance isNoetherian_pi' {R ι M : Type*} [Ring R] [AddCommGroup M] [Module R M
 
 end
 
+section CommRing
+
+variable (R M N : Type*) [CommRing R] [AddCommGroup M] [AddCommGroup N] [Module R M] [Module R N]
```

Comment:
Indeed I was aware of this but the result cannot even be stated in today's Mathlib without commutativity because we only have an `R`-module structure on `N →ₗ[R] M` when `R` is commutative.

So I claim this is an independent concern.

Reply:
What about [LinearMap.module](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/LinearMap.html#LinearMap.module)

Reply:
That would only work if the statement were something like `IsNoetherian R (N →ₗ[S] M)`

Reply:
It is [LinearMap.module](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/LinearMap.html#LinearMap.module) which demands commutativity, via the `SMulCommClass` argument.

You can witness this as follows:
```lean
variable {R M N : Type*} [Ring R] [AddCommGroup M] [Module R M] [AddCommGroup N] [Module R N]

-- Error: "failed to synthesize instance `SMulCommClass R R N`" (disappears if use `CommRing R`)
#check @LinearMap.module R R R M N _ _ _ _ _ _ _ _ _ _
```

Reply:
Oh, I see. Sorry for the useless noise…